opendevin/sandbox/sandbox.py

```diff
@@ -165,10 +164,26 @@ def execute_in_background(self, cmd: str) -> BackgroundCommand:
         self.cur_background_id += 1
         return bg_cmd
 
+    def get_pid(self, bg_cmd):
+        exec_result = self.container.exec_run("ps aux")
+        processes = exec_result.output.decode('utf-8').splitlines()
+        cmd = " ".join(self.get_exec_cmd(bg_cmd.command))
+
+        for process in processes:
+            if cmd in process:
+                pid = process.split()[1] # second column is the pid
+                return pid
+        return None
+
     def kill_background(self, id: int) -> BackgroundCommand:
         if id not in self.background_commands:
             raise ValueError("Invalid background command id")
         bg_cmd = self.background_commands[id]
+        pid = self.get_pid(bg_cmd)
```

Comment:
should probably either remove the `kill` function on BackgroundCommand, or move this logic there

Would be nice if we had the `pid` available in the BackgroundCommand object

Reply:
Done